File: backend/src/main/java/com/task/util/Labels.java
package com.task.util;

import java.util.Map;

/**
 * P2.2: 共享的角色 / 任务状态 / 优先级 标签常量
 *
 * <p>前端 {@code frontend/src/utils/labels.ts} 字段与本类完全一致，
 * 修改任一处需同步另一处。</p>
 *
 * @author Mavis
 */
public final class Labels {

    private Labels() {
        // utility class
    }

    // ============================================
    // 角色
    // ============================================

    public static final String ROLE_EMPLOYEE = "EMPLOYEE";
    public static final String ROLE_MANAGER = "MANAGER";
    public static final String ROLE_ADMIN = "ADMIN";

    public static final Map<String, String> ROLE_LABELS = Map.of(
            ROLE_EMPLOYEE, "普通员工",
            ROLE_MANAGER, "部门经理",
            ROLE_ADMIN, "超级管理员"
    );

    // ============================================
    // 任务状态
    // ============================================

    public static final String STATUS_PENDING_ACCEPT = "PENDING_ACCEPT";
    public static final String STATUS_IN_PROGRESS = "IN_PROGRESS";
    public static final String STATUS_PENDING_VERIFY = "PENDING_VERIFY";
    public static final String STATUS_COMPLETED = "COMPLETED";
    public static final String STATUS_WITHDRAWN = "WITHDRAWN";
    public static final String STATUS_REJECTED = "REJECTED";

    public static final Map<String, String> STATUS_LABELS = Map.of(
            STATUS_PENDING_ACCEPT, "待接收",
            STATUS_IN_PROGRESS, "进行中",
            STATUS_PENDING_VERIFY, "待验收",
            STATUS_COMPLETED, "已完成",
            STATUS_WITHDRAWN, "已撤回",
            STATUS_REJECTED, "已驳回"
    );

    // ============================================
    // 优先级
    // ============================================

    public static final int PRIORITY_HIGHEST = 1;
    public static final int PRIORITY_HIGH = 2;
    public static final int PRIORITY_MEDIUM = 3;
    public static final int PRIORITY_LOW = 4;

    /**
     * 根据优先级数值返回中文标签
     *
     * @param priority 1=最高 2=高 3=中 4=低
     * @return 中文标签；未知值返回 "中"
     */
    public static String getPriorityLabel(Integer priority) {
        if (priority == null) {
            return "中";
        }
        switch (priority) {
            case PRIORITY_HIGHEST:
                return "最高";
            case PRIORITY_HIGH:
                return "高";
            case PRIORITY_MEDIUM:
                return "中";
            case PRIORITY_LOW:
                return "低";
            default:
                return "中";
        }
    }

    /**
     * 根据角色返回中文标签
     */
    public static String getRoleLabel(String role) {
        if (role == null) {
            return ROLE_LABELS.get(ROLE_EMPLOYEE);
        }
        return ROLE_LABELS.getOrDefault(role, role);
    }

    /**
     * 根据任务状态返回中文标签
     */
    public static String getStatusLabel(String status) {
        if (status == null) {
            return "";
        }
        return STATUS_LABELS.getOrDefault(status, status);
    }
}

File: frontend/src/utils/labels.ts
/**
 * P2.2: 共享的角色/状态/优先级 常量与标签
 *
 * <p>所有 *.vue 文件应从这里 import，不要在本地重新定义。
 * 字段名 / 数值 与后端 {@code com.task.util.Labels} 完全一致。</p>
 *
 * <p>状态值对应 {@code com.task.entity.Task.status}：</p>
 * <ul>
 *   <li>PENDING_ACCEPT — 待接收</li>
 *   <li>IN_PROGRESS — 进行中</li>
 *   <li>PENDING_VERIFY — 待验收</li>
 *   <li>COMPLETED — 已完成</li>
 *   <li>WITHDRAWN — 已撤回</li>
 *   <li>REJECTED — 已驳回</li>
 * </ul>
 *
 * <p>角色对应 {@code com.task.entity.User.role}：</p>
 * <ul>
 *   <li>EMPLOYEE — 普通员工</li>
 *   <li>MANAGER — 部门经理</li>
 *   <li>ADMIN — 超级管理员</li>
 * </ul>
 *
 * <p>优先级对应 {@code Task.priority} (1=最高 2=高 3=中 4=低)：</p>
 */

export const TASK_STATUS = {
  PENDING_ACCEPT: 'PENDING_ACCEPT',
  IN_PROGRESS: 'IN_PROGRESS',
  PENDING_VERIFY: 'PENDING_VERIFY',
  COMPLETED: 'COMPLETED',
  WITHDRAWN: 'WITHDRAWN',
  REJECTED: 'REJECTED'
} as const

export type TaskStatus = typeof TASK_STATUS[keyof typeof TASK_STATUS]

export const STATUS_LABELS: Record<TaskStatus, string> = {
  PENDING_ACCEPT: '待接收',
  IN_PROGRESS: '进行中',
  PENDING_VERIFY: '待验收',
  COMPLETED: '已完成',
  WITHDRAWN: '已撤回',
  REJECTED: '已驳回'
}

export const ROLE = {
  EMPLOYEE: 'EMPLOYEE',
  MANAGER: 'MANAGER',
  ADMIN: 'ADMIN'
} as const

export type Role = typeof ROLE[keyof typeof ROLE]

export const ROLE_LABELS: Record<Role, string> = {
  EMPLOYEE: '普通员工',
  MANAGER: '部门经理',
  ADMIN: '超级管理员'
}

/**
 * Element Plus el-tag type，用于 ADMIN 红、MANAGER 黄、EMPLOYEE 灰
 */
export const ROLE_TAG_TYPES: Record<Role, '' | 'success' | 'warning' | 'info' | 'danger'> = {
  EMPLOYEE: 'info',
  MANAGER: 'warning',
  ADMIN: 'danger'
}

/** 优先级数值常量（与后端 Task 实体 @Min/@Max 一致） */
export const PRIORITY = {
  HIGHEST: 1,
  HIGH: 2,
  MEDIUM: 3,
  LOW: 4
} as const

export type Priority = typeof PRIORITY[keyof typeof PRIORITY]

export const PRIORITY_LABELS: Record<Priority, string> = {
  [PRIORITY.HIGHEST]: '最高',
  [PRIORITY.HIGH]: '高',
  [PRIORITY.MEDIUM]: '中',
  [PRIORITY.LOW]: '低'
}

export const PRIORITY_TAG_TYPES: Record<Priority, '' | 'success' | 'warning' | 'info' | 'danger'> = {
  [PRIORITY.HIGHEST]: 'danger',
  [PRIORITY.HIGH]: 'warning',
  [PRIORITY.MEDIUM]: '',
  [PRIORITY.LOW]: 'info'
}
